### PR TITLE
add http client for registering and unregistering with hub

### PR DESCRIPTION
**Pull request by charliesignalfx**
* add http client for registering and unregistering from the hub
* define easyjson data structures for interacting with the hub api's

### CODE REVIEW

File: hub/httpclient/client.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+			authenticationToken: h.authToken,
+		}
+
+		// make the request
+		var respHeaders http.Header
+		var statusCode int
+		var respBody []byte
+		statusCode, respHeaders, respBody, err = h.request(registerV2, http.MethodPost, reqBody, headers)
+		if err == nil {
+			etag = respHeaders.Get("etag")
```
Comment (sstewart-signalfx) [inline hub/httpclient/client.go]: Should "etag" be included in the header constants defined above?

Comment (sstewart-signalfx) [inline hub/httpclient/client.go]: I think the header is "ETag", instead of "etag"

File: hub/hubclient/hubclient.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	Name string `json:"name"`
+	// Version is the smart gateway version of the server
+	Version string `json:"version"`
+	// Payload is the payload for the server
+	Payload []byte `json:"payload"`
+	// LastHeartbeat is the timestamp of the last heartbeat
+	LastHeartbeat int `json:"lastHeartbeat"`
+}
+
+// String is the too ToString method for Server
```
Comment (sstewart-signalfx) [inline hub/hubclient/hubclient.go]: "too ToString" - I think there's an extra word in the comment

Comment (charliesignalfx) [reply hub/hubclient/hubclient.go]: whoops yeah that's not right

**Pull request merged**